lib/flat_search_web/live/page_live.ex

```diff
@@ -13,7 +13,46 @@ defmodule FlatSearchWeb.PageLive do
 
   @impl true
   def handle_info({:flat_created, flat}, socket) do
-    {:noreply, assign(socket, flats: [flat] ++ socket.assigns.flats)}
+    insensitive_string("Warszawą Dąbrowa")
+
+    case flat_fulfills_filters?(socket.assigns.filters, flat) do
+      true ->
+        {:noreply, assign(socket, flats: [flat] ++ socket.assigns.flats)}
+
+      false ->
+        {:noreply, assign(socket, flats: socket.assigns.flats)}
+    end
+  end
+
+  defp flat_fulfills_filters?(user_filters, flat_filters) do
+    # compare only if key exists in flat_filters and user_filters key is not empty
+    # enum return ok?
+    key_exists_and_has_value? =
+      Enum.each(user_filters, fn
+        {_key, none} when none in ["", nil] ->
+          false
+
+        {key, _value} ->
+          Map.has_key?(flat_filters, String.to_existing_atom(key))
+      end)
```

Comment:
Enum.each  always return :ok 😛  use Enum.any? maybe, to check if at least one in the enumerator passes the condition, or Enum.all? if all should pass this condition